channel.go

```diff
@@ -0,0 +1,44 @@
+// unbounded buffered channel implementation
+// inspired by https://medium.com/capital-one-tech/building-an-unbounded-channel-in-go-789e175cd2cd
+
+package sequence
+
+import "github.com/0xsequence/go-sequence/lib/logadapter"
+
+// converts a blocking unbuffered send channel into a non-blocking unbounded buffered one
+func makeUnboundedBuffered(sendCh chan<- *ReceiptResult, log logadapter.Logger, bufferLimitWarning int) chan<- *ReceiptResult {
```

Comment:
I think we should move this method to the receipt_listener.go file at the very bottom.. since its for receipt results and the only place its really used